### PR TITLE
[stdlib][nfc] Move (most) SDK tests into their own folder.

### PR DESCRIPTION
**Pull request by airspeedswift**
Similar to how they are in source. Should help clear up confusion about where to route issues. Also renames a handful of tests that were prefixed with `Test`, as this pattern isn't consistently followed.

### CODE REVIEW

**Comment by airspeedswift** [comment]
@swift-ci please test

**Review by lancep — APPROVED** [review]
LGTM!

**Comment by jrose-apple** [comment]
That HomeKit failure is valid—you're shadowing the HomeKit overlay.

**Comment by airspeedswift** [comment]
Ah, I'm skipping iOS tests when I ran them locally, duh.

**Review by parkera — APPROVED** [review]

File: validation-test/SDK/Inputs [hunk 1]
```diff
@@ -0,0 +1 @@
+../../test/sdk/Inputs
```
Comment (benrimmington) [inline validation-test/SDK/Inputs]: Should this symbolic link be `../../test/SDK/Inputs` for case-sensitive volumes?

**Comment by moiseev** [comment]
While we're at it, I think the overlay tests that are in `validation-test` should not be there, and are most of the time just "-typecheck -verify" tests. Why don't we just move them all in either `test` or `validation-test`? Since our CI would currently run validation tests for macOS even in "smoke test" mode, I'm slightly in favor of `validation-test/SDK` as a final destination.

**Comment by jrose-apple** [comment]
Please don't. Validation tests are over there because they slow down local testing. Ideally everything would be in test/, but in practice the signal-to-(time-and-noise) ratio is low for a handful of things.

**Comment by futurejones** [comment]
Can one of the admins verify this patch?

**Comment by shahmishal** [comment]
Please update the base branch to `main` by Oct 5th otherwise the pull request will be closed automatically. 

* How to change the base branch: ([Link](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/changing-the-base-branch-of-a-pull-request))
* More detail about the branch update: ([Link](https://forums.swift.org/t/updating-branch-names/40412/))

**Pull request closed without merging**